rag/svr/task_broker.py

```diff
@@ -26,111 +26,109 @@
 from deepdoc.parser.excel_parser import RAGFlowExcelParser
 from rag.settings import cron_logger
 from rag.utils.minio_conn import MINIO
-from rag.utils import findMaxTm
-import pandas as pd
 from api.db import FileType, TaskStatus
 from api.db.services.document_service import DocumentService
 from api.settings import database_logger
 from api.utils import get_format_time, get_uuid
-from api.utils.file_utils import get_project_base_directory
 from rag.utils.redis_conn import REDIS_CONN
 from api.db.db_models import init_database_tables as init_web_db
 from api.db.init_data import init_web_data
+from rag.settings import SVR_QUEUE_NAME
+from rag.svr.task_executor import run_embedding
+from concurrent.futures import ThreadPoolExecutor
+POOL_EXECUTOR = ThreadPoolExecutor(max_workers=5)
 
 
-def collect(tm):
-    docs = DocumentService.get_newly_uploaded(tm)
-    if len(docs) == 0:
-        return pd.DataFrame()
-    docs = pd.DataFrame(docs)
-    mtm = docs["update_time"].max()
-    cron_logger.info("TOTAL:{}, To:{}".format(len(docs), mtm))
-    return docs
-
-
-def set_dispatching(docid):
+def dispatching(docid):
     try:
         DocumentService.update_by_id(
             docid, {"progress": random.random() * 1 / 100.,
                     "progress_msg": "Task dispatched...",
                     "process_begin_at": get_format_time()
                     })
+        POOL_EXECUTOR.submit(run_embedding, docid)
     except Exception as e:
         cron_logger.error("set_dispatching:({}), {}".format(docid, str(e)))
 
 
 def dispatch():
-    tm_fnm = os.path.join(
-        get_project_base_directory(),
-        "rag/res",
-        f"broker.tm")
-    tm = findMaxTm(tm_fnm)
-    rows = collect(tm)
-    if len(rows) == 0:
+    try:

```

Comment:
As designed， broker is used for dispatching task and updating progress, and executor is for processing task including embedding since we can deploy one executor or multiple executors in different machines to enlarge the document processing through output.
I suggest that:
1. remove broker, and put the progress updating function to ragflow_server using multi-thread.
2. the executor reads tasks from redis queue and execute them.

Reply:
Yeap, I'm doing this suggest.
The goal is to make the design simpler.